tests/test_taylor_coefs_one_column_grp.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+import pandas as pd
+import numpy as np
+from timeseriesutils.featurize import taylor_coefs_one_column_grp
+
+
+def mean_fn(t, a, b, phi=10, d=0):
+  '''
+  mean function, or its derivative, used for simulating data
+  the function is mu(t) = a * cos(t/phi) + b * sin(t/phi)
+  the first derivative is mu'(t) = -a / phi * sin(t/phi) + b / phi * cos(t/phi)
+  the second derivative is mu''(t) = -a / phi**2 * cos(t/phi) - b / phi**2 * sin(t/phi)
+  
+  Parameters
+  ----------
+  t: numpy array of time points at which to evaluate the function
+  a, b: coefficients for cos and sin terms
+  phi: scaling factor applied to t
+  d: integer derivative to evaluate: 0, 1, or 2
+  
+  Returns
+  -------
+  numpy array of same length as t with function values
+  '''
+  if d == 0:
+    return a * np.cos(t / phi) + b * np.sin(t / phi)
+  elif d == 1:
+    return -a / phi * np.sin(t / phi) + b / phi * np.cos(t / phi)
+  elif d == 2:
+    return -a / (phi**2) * np.cos(t / phi) - b / (phi**2) * np.sin(t / phi)
+
+
+def one_grp_ex(a=1, b=0.2):
+  '''
+  Generate example data for one "group", e.g. one location
+  
+  Parameters
+  ----------
+  a, b: cos and sin coefficients
+  
+  Returns
+  -------
+  Pandas data frame with columns `t` with a time index and
+  `y` with response variable values
+  '''
+  rng = np.random.default_rng(12345)
+  t = np.arange(50)
+  return pd.DataFrame({
+    't': t,
+    'y': rng.normal(mean_fn(t, a=a, b=b), scale=0.1)
+  })
+
+
+def test_taylor_coefs_one_column_grp_trailing():
+  df = taylor_coefs_one_column_grp(
+    data=one_grp_ex(),
+    c='y',
+    taylor_degree=2,
+    window_size=14,
+    window_align='trailing',
+    ew_span=None,
+    fill_edges=False)
+  
+  mean = mean_fn(t=df['t'], a=1, b=0.2, d=0)
+  mean_d1 = mean_fn(t=df['t'], a=1, b=0.2, d=1)
+  mean_d2 = mean_fn(t=df['t'], a=1, b=0.2, d=2)
+  
+  new_cols = [f'y_taylor_d2_c{c}_w14t_sNone' for c in range(3)]
+  
+  assert list(df.columns) == ['t', 'y'] + new_cols
+  assert np.all(np.isnan(df[new_cols].iloc[:14]))
+  assert np.all(~np.isnan(df[new_cols].iloc[14:]))
+  assert df[new_cols[0]].iloc[14:].values == pytest.approx(mean[14:], abs = 0.2)
+  assert df[new_cols[1]].iloc[14:].values == pytest.approx(mean_d1[14:], abs = 0.1)
+  assert df[new_cols[2]].iloc[14:].values == pytest.approx(mean_d2[14:], abs = 0.02)
```

Comment:
One last plug for a pytest feature: parameterized tests. I'm not sure if it's appropriate here (tbh, I have no idea what `taylor_coefs_one_column_grp` is supposed to do).

If you find that you want to test this function with a variety of inputs without writing repetitive boilerplate, you can parameterize the test: given a list of inputs and expected outputs, run the test multiple times.

Even if it's not useful for this specific test, here's what this might look like (in this example, I also factored out some of the helper functions, since DRY can sometimes interfere with test readability (reasonable people can disagree about where the line is):

```
import pytest
import pandas as pd
import numpy as np
from timeseriesutils.featurize import taylor_coefs_one_column_grp


@pytest.fixture()
def one_grp_ex(a=1, b=0.2):
  '''
  Generate example data for one "group", e.g. one location
  
  Parameters
  ----------
  a, b: cos and sin coefficients
  
  Returns
  -------
  Pandas data frame with columns `t` with a time index and
  `y` with response variable values
  '''
  rng = np.random.default_rng(12345)
  t = np.arange(50)
  df =  pd.DataFrame({
    't': t,
    'y': rng.normal((a * np.cos(t / 10) + b * np.sin(t / 10)), scale=0.1),
  })
  return df
  

@pytest.fixture()
def means(one_grp_ex) -> dict:
  a = 1
  b = 0.2
  phi = 10
  t = one_grp_ex['t']
  mean_fn = {}
  mean_fn[0] = a * np.cos(t / phi) + b * np.sin(t / phi)
  mean_fn[1] = -a / phi * np.sin(t / phi) + b / phi * np.cos(t / phi)
  mean_fn[2] = -a / (phi**2) * np.cos(t / phi) - b / (phi**2) * np.sin(t / phi)
  return mean_fn


@pytest.mark.parametrize(
  'derivative,approx_abs',
  [
    (0, 0.2),
    (1, 0.1),
    (2, 0.02)
  ]
)
def test_with_parameters(means, one_grp_ex, derivative, approx_abs):
    df = taylor_coefs_one_column_grp(
      data=one_grp_ex,
      c='y',
      taylor_degree=2,
      window_size=14,
      window_align='trailing',
      ew_span=None,
      fill_edges=False)
    
    new_col = f'y_taylor_d2_c{derivative}_w14t_sNone'
    
    assert np.all(np.isnan(df[new_col].iloc[:14]))
    assert np.all(~np.isnan(df[new_col].iloc[14:]))
    assert df[new_col].iloc[14:].values == pytest.approx(means[derivative][14:], abs = approx_abs)
```